### ISSUE TITLE
the dataloader is very slow if when using the patch sampling of points and normals

### ISSUE THREAD

**Issue by sinAshish** [lingtengqiu/Open-PIFuhd#19]
how can I speed up the training process because of the aforementioned issue.  I have not cached the sampled points because I want to keep the randomness! Any solutions for this? 
@lingtengqiu

**Comment by lingtengqiu** [comment]
considering the training speed, we sample 500000 points in cache files. I think it is enough to keep training randomness. If you want to online sample training points, I suggest you use an occupancy-check implemented based on GPU~(e.g., Kaolin). However, online checking also spends 10x times on sampling.

**Comment by sinAshish** [comment]
Thank you!. I’ll check out kaolin.

Currently, I’m also doing offline sampling of points, and it’s much faster. While for occupancy checking, I was using libmesh package in Occupancy Networks repo.
The bottle neck for my code is the querying of points in the grid from the implicit function before performing marching cubes. Since, I am using the gradients of the the occupancy grid in my computation graph, each iteration is very expensive wrt time and memory.
